tests/django_pg_migration_tools/test_operations.py

```diff
@@ -427,3 +454,355 @@ def test_when_not_allowed_to_migrate(self):
+            # re-creating the unique index from scratch.
+            cursor.execute(_CREATE_UNIQUE_INDEX_QUERY)
+            cursor.execute(_SET_INDEX_INVALID, {"index_name": "unique_int_field_idx"})
+            # Also, set the lock_timeout to check it has been returned to
+            # its original value once the unique index creation is completed.
+            cursor.execute(_SET_LOCK_TIMEOUT)
+
+        # Prove that the invalid unique index exists before the operation runs:
+        with connection.cursor() as cursor:
+            cursor.execute(
+                operations.SaferAddUniqueConstraint.CHECK_INVALID_INDEX_QUERY,
+                {"index_name": "unique_int_field_idx"},
+            )
+            assert cursor.fetchone()
+
+        # Prove that the constraint does **not** already exist.
+        with connection.cursor() as cursor:
+            cursor.execute(
+                operations.SaferAddUniqueConstraint._CHECK_EXISTING_CONSTRAINT_QUERY,
+                {"constraint_name": "unique_int_field"},
+            )
+            assert not cursor.fetchone()
+
+        project_state = ProjectState()
+        project_state.add_model(ModelState.from_model(IntModel))
+        new_state = project_state.clone()
+
+        operation = operations.SaferAddUniqueConstraint(
+            model_name="intmodel",
+            constraint=UniqueConstraint(
+                fields=("int_field",),
+                name="unique_int_field",
+            ),
+        )
+
+        assert operation.describe() == (
+            "Concurrently adds a UNIQUE index unique_int_field_idx on model intmodel "
+            "on field(s) ('int_field',) if the index does not exist. Then, adds the "
+            "constraint using the just-created index. NOTE: "
+            "Using django_pg_migration_tools SaferAddUniqueConstraint operation."
+        )
+
+        name, args, kwargs = operation.deconstruct()
+        assert name == "SaferAddUniqueConstraint"
+        assert args == []
+        assert kwargs == {"model_name": "intmodel", "constraint": operation.constraint}
+
+        operation.state_forwards(self.app_label, new_state)
+        assert (
+            len(new_state.models[self.app_label, "intmodel"].options["constraints"])
+            == 1
+        )
+        assert (
+            new_state.models[self.app_label, "intmodel"].options["constraints"][0].name
+            == "unique_int_field"
+        )
+
+        # Proceed to add the unique index followed by the constraint:
+        with connection.schema_editor(atomic=False, collect_sql=False) as editor:
+            with utils.CaptureQueriesContext(connection) as queries:
+                operation.database_forwards(
+                    self.app_label, editor, project_state, new_state
+                )
+
+        # Assert the index has disappeared!!!
+        # Be no alarmed, this seems a bit strange, but Postgres has "swapped"
```

Comment:
should this be "Be not alarmed"?

Reply:
It should (-: fixup: 3ec5aea02bfe2b24c3a3f864e368eae518fdc219